crates/bevy_scene/src/serde.rs

```diff
@@ -812,9 +812,15 @@ mod tests {
                 let received = received
                     .components
                     .iter()
-                    .find(|component| component.type_name() == expected.type_name())
+                    .find(|component| {
+                        component.get_represented_type_info().unwrap().type_path()
+                            == expected.get_represented_type_info().unwrap().type_path()
```

Comment:
Minor optimization: Can we move `expected.get_represented_type_info()` out of this closure to prevent calling it repeatedly?

Reply:
its a test-only fn so its not performance critical but i don't see why not!

Reply:
Lol my bad I completely missed the context of this being a test